src/persist-client/src/lib.rs

```diff
@@ -351,6 +353,27 @@ impl PersistClient {
         let mut machine = self.make_machine(shard_id, diagnostics.clone()).await?;
         let gc = GarbageCollector::new(machine.clone(), Arc::clone(&self.isolated_runtime));
 
+        // TODO: Feels like we could combine this into the
+        // `register_leased_reader` call once this finishes rolling out.
+        let schema_id = if SCHEMA_REGISTER.get(&self.cfg) {
+            // NB: The overwhelming common case is that this schema is already
+            // registered. In this case, the cmd breaks early and nothing is
+            // written to (or read from) CRDB.
+            let (schema, maintenance) = machine.register_schema(&*key_schema, &*val_schema).await;
```

Comment:
I think we only want to register schemas when opening a `WriteHandle` not a `ReadHandle`. 

@jkosh44 and I chatted about a Pv2 workflow for migrating builtin tables where a new read-only environment comes online that adds a column to a builtin table. We want to be able to read as though the column is there, i.e. fill in default values, but we don't want the new schema for the builtin table to get registered in-case the upgrade fails.

I think this should be okay because AFAIU only `WriteHandle`s actually commit new data, either through new writes or compaction, so if a `ReadHandle` is opened with a new schema that shouldn't impact the state of the data.

We could always start by not registering when opening a `ReadHandle` and add it in later? What do you think?

Reply:
Yeah, fair enough. I think we probably do want to look up a schema id (when it exists) in the reader path, but that can definitely wait until we get further along in schema change work. Removed